### DISCUSSION TITLE
Propogating SecurityContext to job

### DISCUSSION THREAD

**Discussion by igorg-il** [jobrunr/jobrunr#396] [Q&A]
Is there as way to propogate security context to jobrunr job so we can get logged-in user name?

There is a way to propogate it to Async. Is there something similar for jobrunr?
https://baeldung-cn.com/spring-security-async-principal-propagation

Currently my only solution is to pass user name as job argument.
thank you

**Comment by patkujawa-wf** [comment]
How would that even be possible? The job may run on an entirely different machine, so the context would need to be persisted in the db somewhere. Perhaps you are just looking for a more automated way to include that info, e.g. with annotations on the job?

**Reply by igorg-il** [reply]
Yes. Makes sense that it should be persisted.
